docs/src/cli-reference/TensileCreateLibrary.rst

```diff
@@ -0,0 +1,133 @@
+
+\-\-architecture=ARCHITECTURE
+    Architectures to generate a library for. When specifying multiple options, use quoted, semicolon delimited 
+    architectures, e.g., --architecture='gfx908;gfx1012'.
+    Supported architechures include: all gfx000 gfx803 gfx900 gfx900:xnack- gfx906 gfx906:xnack+ gfx906:xnack- gfx908 gfx908:xnack+
+    gfx908:xnack- gfx90a gfx90a:xnack+ gfx90a:xnack- gfx940 gfx940:xnack+ gfx940:xnack- gfx941 gfx941:xnack+
+    gfx941:xnack- gfx942 gfx942:xnack+ gfx942:xnack- gfx1010 gfx1011 gfx1012 gfx1030 gfx1031 gfx1032 gfx1034 gfx1035
+    gfx1100 gfx1101 gfx1102.
+\-\-build-client
+    Build Tensile client executable; used for stand alone benchmarking (default).
+\-\-client-config 
+    Creates best-solution.ini in the output directory for the library and code object files created (default).
+\-\-code-object-version={default,V4,V5}
+    HSA code-object version.
+\-\-cxx-compiler={hipcc}
+    C++ compiler used when generating binaries.
+\-\-embed-library=EMBEDLIBRARY
+    Embed (new) library files into static variables. Specify the name of the library.
+\-\-embed-library-key=EMBEDLIBRARYKEY
+    Access key for embedding library files.
+\-\-generate-manifest-and-exit
+    Similar to dry-run option for *make*, will compute the outputs
+    of *TensileCreateLibrary* and write the expected outputs to a 
+    manifest file but does not exectue the commands to generate the 
+    output.
+\-\-generate-sources-and-exit
+    Skip building source code object files and assembly code object files.
+    Output source files only and exit. 
+\-\-ignore-asm-cap-cache
+    Ignore asm capability cache and derive the asm capabilities at runtime.    
+\-\-jobs=CPUTHREADS, \-j CPUTHREADS
+    Number of parallel jobs to launch. If this options is set higher than *nproc* the number of parallel 
+    jobs will be equal to the number of cores. If the this option is set below 1 (e.g. 0 or -1), the number
+    of parallel jobs will be set to the number of cores, up to a maximum of 64. (default = -1).    
+\-\-lazy-library-loading
+    Loads Tensile libraries when needed instead of upfront.
+\-\-library-format={yaml,msgpack}
+    Select which library format to use (default: msgpack).
+\-\-library-print-debug
+    Solutions will print enqueue info when enqueueing a kernel.
+\-\-no-enumerate
+    Do not run rocm_agent_enumerator.
+\-\-no-library-print-debug
+    Solutions will not print enqueue info when enqueueing a kernel (default).
+\-\-no-merge-files
+    Store every solution and kernel in separate file.
+\-\-no-short-file-names
+    Disables short files names.
+\-\-num-merged-files=NUMMERGEDFILES
+    Number of files the kernels should be written into.
+\-\-merge-files
+    Store all solutions in single file (default).
+\-\-package-library 
+    Enable the packaging of the client library code objects into separate (deprecated).
+    architectures as sub-directories within the Tensile/library directory.
```

Comment:
```suggestion
\-\-package-library 
    Enable the packaging of the client library code objects into separate
    architectures as sub-directories within the Tensile/library directory (deprecated).
```

Reply:
Thoughts on noting that undocumented options are not considered supported?

Reply:
I think it's a good idea, we should also then add `deprecated` to the help docs directly.